### ISSUE TITLE
Detect RCODE-INFO:DISPLAY-TYPE?

### ISSUE THREAD

**Issue by braun1928** [Riverside-Software/pct#246]
Hello,
I've started fooling around, and it's awesome! I sill haven't found one item, though. Our deployment is mixed, with most r-codes running in Windows 32 GUI, but some of them run on Linux server (_progres or AppServer). Because of this, some r-code have to be compiled as TTY. Today I check DISPLAY-TYPE and the extension (ignore .w), and if it is GUI, I run a compile statement on Linux (though an AppServer) to have it in TTY.
How should one achieve this with PCT? The first thought was renaming needed files, but it will be a long time doing that. Or recompile all possible files, even though I'd have two r-codes even when only one would be needed (DISPLAY-TYPE eq ''). Ideally I'd do the same as I do today, compiling in TTY only when needed, and saving into another directory, of course, but I haven't found any documentation regarding this situation.
Thanks!

**Comment by gquerret** [comment]
You can have two different PCTCompile tasks, first one with graphicalMode="true" and everything in your filesets, and a second one with graphicalMode="false", another destDir, and everything but the .w files in your filesets (`<fileset dir="xxx" excludes="**/*.w" />` for example).

**Comment by braun1928** [comment]
Yes, that's what I'm doing so far. But currently less than 7% of the r-code *needs* TTY, that's why I was wondering about checking DISPLAY-TYPE to compile only what's really needed -- thus, having a smaller compiler time, and fewer compiled objects. :)
If it's not possible (or it's not a common problem, at least), I'll stick to compile everything in both environments.

**Comment by gquerret** [comment]
Could be an enhancement request for [RCodeSelector](https://github.com/Riverside-Software/pct/wiki/RCodeSelector), but that's currently not implemented however

**Comment by gquerret** [comment]
I had a discussion some time ago with Progress, and this information is not available directly from the r-code.

**Comment by stefandrissen** [comment]
Just saw this one again and would like to share my 2 cents. A long time ago we came from a similar situation with a fat client environment in which everything was compiled with a graphical client. We started separating parts out to run on an AppServer and WebSpeed. For the first versions we just compiled everything with both _progres and prowin32. At first ignoring all errors being thrown by _progres and later ignoring certain errors and adding other bits of nonsense to the files. 

Later on we had pretty much everything available on the AppServer and had only a thin set of sources for the WebClient - which files were compile with prowin32 and _progres was all a bit magical. First everything was compile with _progres, everything that did not compile or matched a certain pattern (*-c.p) was compiled with prowin32. To make sure files that could compile on _progres but were irrelevant for the server side we had ForceErrorInChUI preprocessor. And there was also a hard-coded file list floating around somewhere.

This actually all worked fine. 

Except when new people came in and you tried explaining this to them.

So I separated the whole lot out with a top level of folders indicating, the role, simplified this was:

ui
server
i

The i folder contained includes that can be used by both.

Believe me, it helps the fight against insanity. :-)
